### PR TITLE
Add test coverage of the GGUFWriter class by adding several test cases.

### PR DESCRIPTION
**Pull request by mendax0110**
**Description:**

This pull request aims to enhance the test coverage of the GGUFWriter class by adding several test cases. 

**Changes Made:**

The following test cases have been added:

1. `test_gguf_writer`: This test case creates an instance of GGUFWriter, adds metadata, architecture information, and a tensor, and then writes the data to a file. Finally, it verifies the existence of the generated file.

2. `test_ggml_quantization_type`: This test case checks the correctness of GGMLQuantizationType constants, ensuring they have the expected values.

3. `test_gguf_value_type`: Similar to the previous test case, this one checks the correctness of GGUFValueType constants.

**Rationale:**

By adding these test cases, we improve the reliability of the GGUFWriter class. 
With these tests in place, we can have more confidence in the correctness of the GGUF library.

**Additional Notes:**

- All tests have been designed to clean up after themselves using a temporary directory created with `tempfile.TemporaryDirectory()`.
- The tests include assertions to verify the expected outcomes, making them suitable for use in a continuous integration (CI) environment.

## Reviewers

Any feedback on the added test cases and their coverage is greatly appreciated.

### CODE REVIEW

File: gguf-py/tests/test_gguf.py [hunk 1]
```diff
@@ -1,7 +1,48 @@
...
+    tensor_data = np.random.rand(10, 10).astype(np.float32)
+    writer.add_tensor("test_tensor", tensor_data)
+
+    # Write to file and close
+    writer.write_header_to_file()
+    writer.write_tensors_to_file()
+    writer.close()
+
+    # Verify the generated file exists
+    assert os.path.exists(os.path.join(temp_dir, "test.gguf"))
```
Comment (akawrykow) [inline gguf-py/tests/test_gguf.py:36]: Should we also verify the contents of the file? You could either create this actual file as a  gguf with the expected contents and check it in then compare whatever file we write with the snapshot, or you could use a gguf reader and pull out all the keys and assert they have the right values. That will guard against inadvertent changes to the format etc

File: gguf-py/tests/test_gguf.py [hunk 2]
```diff
@@ -1,7 +1,48 @@
...
+
+# Test GGMLQuantizationType
+def test_ggml_quantization_type():
+    assert GGMLQuantizationType.F32 == 0
+    assert GGMLQuantizationType.F16 == 1
+    assert GGMLQuantizationType.Q4_0 == 2
```
Comment (Cebtenzzre) [inline gguf-py/tests/test_gguf.py:42]: This test case doesn't do anything useful.

File: gguf-py/tests/test_gguf.py [hunk 3]
```diff
@@ -1,7 +1,48 @@
...
+
+# Test GGUFValueType
+def test_gguf_value_type():
```
Comment (monatis) [inline gguf-py/tests/test_gguf.py:45]: This test is also redundant. There's no need to write test cases for enums / constants.
A better test case would be to read the written GGUF file, but there's no reader class implemented in Python yet. So we can verify the checksum of the file instead.

**Review by monatis — CHANGES_REQUESTED** [review]

**Comment by Cebtenzzre** [comment]
@mendax0110 There's no need to push merges if there aren't any conflicts. Every time you merge master, I get a notification.

**Comment by mendax0110** [comment]
> @mendax0110 There's no need to push merges if there aren't any conflicts. Every time you merge master, I get a notification.

Sorry!

**Comment by alitariq4589** [comment]
Can one of the admins verify this patch?

**Pull request closed without merging**